workshop/exercise-6/README.md

```diff
@@ -0,0 +1,86 @@
+
+A `Secret` is a Kubernetes object containing sensitive data such as a password, a token, or a key. You can also read more about [Secrets](https://kubernetes.io/docs/concepts/configuration/secret/).
+
+1. Let's create a secret, which will be a `docker-registry` type secret. This type of secret is used to authenticate with a container registry to pull down a private image. We can create this via the commandline. For username, simply use the string `iamapikey`. For `<api_key_value>`, use the api key you made note of earlier.
+
+    ```
+    kubectl --namespace default create secret docker-registry ibm-cr-secret  --docker-server=us.icr.io --docker-username=iamapikey --docker-password=$MYAPIKEY
+    ```
+
+2. We will also need a secret for the build process to have credentials to push the built image to the container registry. To create this secret, we'll first need to base64 encode our username and password for IBM Container Registry. For username, you will use the string `iamapikey`. The base64 encoding of `iamapikey` should be: `aWFtYXBpa2V5` - which is already in the yaml file.  Let's base64 encode our apikey, and copy it.
+
+    ```
+    echo -n "$MYAPIKEY" | base64 -w0  # Linux
+    echo -n "$MYAPIKEY" | base64 -b0  # MacOS
+    ```
+
+3. Update the `docker-secret.yaml` file with your base64 encoded password. You can find the password field near the end of the file. Username (`aWFtYXBpa2V5=`) is already provided for you.  Replace `<base_64_encoded_api_key_value>` with your own base64 encoded apikey value, and ensure you've saved the file.
+
+4. Apply the secret to your cluster:
+
+    ```
+    kubectl apply --filename docker-secret.yaml
+    ```
+
+5. A `Service Account` provides an identity for processes that run in a Pod. This Service Account will be used to link the build process for Knative to the Secrets you just created. View the service account file, and notice that it's using the credentials you created earlier for pulling from and pushing to the container registry:
+
+    ```
+    cat service-account.yaml
+    ```
+
+    Example output:
+    ```yaml
+     apiVersion: v1
+     kind: ServiceAccount
+     metadata:
+       name: build-bot
+     secrets:
+     - name: ibm-cr-push-secret
+     imagePullSecrets:
+     - name: ibm-cr-secret
+    ```
+
+
+6. Apply the service account to your cluster:
+
+    ```
+    kubectl apply --filename service-account.yaml
+    ```
+
+Congratulations! You've set up some required credentials that the Tekton pipeline process will use to have access to push to your container registry. In the next exercise, you will build the container, push it to your container registry, and then deploy the app to knative using Tekton Pipelines. The goal of this exercise was to set up some required credentials for that flow.
```

Comment:
s/build the container/build the image/

Reply:
technically, s/container registry/image registry/    but I'm less bothered by that one :-)

Reply:
the service is "IBM Container Registry", so maybe container image registry would work.